compiler-plugin/src/main/java/io/ballerina/lib/np/compilerplugin/PromptAsCodeCodeModificationTask.java

```diff
@@ -362,6 +367,52 @@ public static MappingConstructorExpressionNode getAnnotationExpression(String js
         return (MappingConstructorExpressionNode) NodeParser.parseExpression(jsonSchema);
     }
 
+    private static boolean containsBallerinaxNPImport(NodeList<ImportDeclarationNode> imports) {
+        for (ImportDeclarationNode importDeclarationNode : imports) {
+            Optional<ImportOrgNameNode> importOrgNameNode = importDeclarationNode.orgName();
+            if (importOrgNameNode.isPresent() && importOrgNameNode.get().orgName().text().equals(ORG_NAME)
+                    && importDeclarationNode.moduleName().get(0).text().equals(MODULE_NAME)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static NodeList<ImportDeclarationNode> updateImports(ModulePartNode modulePartNode) {
+        NodeList<ImportDeclarationNode> imports = modulePartNode.imports();
+        NodeList<ModuleMemberDeclarationNode> members = modulePartNode.members();
+        if (containsBallerinaxNPImport(imports)) {
+            return imports;
+        }
+
+        for (ModuleMemberDeclarationNode memberNode : members) {
+            if (memberNode.kind() != SyntaxKind.TYPE_DEFINITION) {
+                continue;
+            }
+
+            TypeDefinitionNode typeDefinitionNode = (TypeDefinitionNode) memberNode;
+            NodeList<AnnotationNode> annotations = getMetadataNode(typeDefinitionNode).annotations();
+            for (AnnotationNode annotation: annotations) {
+                if (isNPSchemaAnnotationAvailable(annotation)) {
+                    return imports.add(createImportDeclarationForNPModule());
+                }
+            }
+        }
+        return imports;
+    }
+
+    public static boolean isNPSchemaAnnotationAvailable(AnnotationNode annotationNode) {
+        if (annotationNode.annotReference() instanceof SimpleNameReferenceNode refNode) {
+            return refNode.name().text().equals(MODULE_NAME + ":" + SCHEMA_ANNOTATION_IDENTIFIER);
+        }
+        return false;
+    }
+
+    public static ImportDeclarationNode createImportDeclarationForNPModule() {
+        return NodeParser.parseImportDeclaration(IMPORT + WS + ORG_NAME + SLASH + MODULE_NAME + WS + AS + WS +
+                                                 MODULE_NAME + SEMI_COLON);
```

Comment:
I would introduce a constant for `import ballerinax/np;` with interpolations for just org and module name.